pkg/registry/garden/shoot/strategy.go

```diff
@@ -139,3 +143,49 @@ func (shootStatusStrategy) PrepareForUpdate(ctx context.Context, obj, old runtim
 func (shootStatusStrategy) ValidateUpdate(ctx context.Context, obj, old runtime.Object) field.ErrorList {
 	return validation.ValidateShootStatusUpdate(obj.(*garden.Shoot).Status, old.(*garden.Shoot).Status)
 }
+
+// ToSelectableFields returns a field set that represents the object
+// TODO: fields are not labels, and the validation rules for them do not apply.
+func ToSelectableFields(shoot *garden.Shoot) fields.Set {
+	// The purpose of allocation with a given number of elements is to reduce
+	// amount of allocations needed to create the fields.Set. If you add any
+	// field here or the number of object-meta related fields changes, this should
+	// be adjusted.
+	shootSpecificFieldsSet := make(fields.Set, 3)
```

Comment:
Why not 1?

Reply:
`generic.AddObjectMetaFieldsSet` adds 2 entries (for namespaced resources).